keepsorted/options.go

```diff
@@ -180,21 +182,50 @@ func parseBlockOptions(commentMarker, options string, defaults blockOptions) (_
 	return ret, warns
 }
 
-func formatValue(val reflect.Value) string {
+func formatValue(val reflect.Value) (string, error) {
 	switch val.Type() {
 	case reflect.TypeFor[bool]():
-		return boolString[val.Bool()]
+		return boolString[val.Bool()], nil
 	case reflect.TypeFor[[]string]():
-		return strings.Join(val.Interface().([]string), ",")
+		return formatList(val.Interface().([]string))
 	case reflect.TypeFor[map[string]bool]():
-		return strings.Join(slices.Sorted(maps.Keys(val.Interface().(map[string]bool))), ",")
+		return formatList(slices.Sorted(maps.Keys(val.Interface().(map[string]bool))))
 	case reflect.TypeFor[int]():
-		return strconv.Itoa(int(val.Int()))
+		return strconv.Itoa(int(val.Int())), nil
 	}
 
 	panic(fmt.Errorf("unsupported blockOptions type: %v", val.Type()))
 }
 
+func formatList(vals []string) (string, error) {
+	var specialChars bool
+	if len(vals) > 0 && vals[0][0] == '[' {
```

Comment:
I'm not sure if it's ever possible for `len(vals) > 0` but `len(vals[0]) == 0` (i.e., `vals` contains an empty string), but probably safest to also check for that too? Don't wanna panic unnecessarily.

An alternative would be to use `strings.HasPrefix`.